app/models/git_repository.rb

```diff
@@ -70,9 +71,14 @@ def self_signed_cert_cb(_valid, _host)
     true
   end
 
+  # NOTE:  Please make sure to use this whenever you implement a method that is
+  # modifying the `.git` data (clone, fetch, checking, etc.) so that we can be
+  # sure this happens safely between processes.
```

Comment:
Rugged handles locking internally, so 2 separate instanced of a Rugged::Repo doing a fetch will not collide with each other.  The only thing we need to handle are clones because that's occurring outside of a "repo" context.

Reply:
I think if there are multiple processes looking at the same repo, I'd also like to *not* serialize them, allowing them to execute git stuff in parallel.

Reply:
Well this would have been good to know prior to doing all this work...

Reply:
Okay, "salty Nick" has left the building...

Does it make sense to then reuse the locking code that is in `GitWorktree`/`Rugged` (honestly still not fully up to speed on which does it currently):

https://github.com/ManageIQ/manageiq/blob/master/lib/git_worktree.rb#L366-L391

or stick with what I have done here?

Reply:
I think the linked locking bits would still have to be after the clone, so it doesn't help us in this case.